### PR TITLE
REGRESSION: fix compilation if !ENABLE(VIDEO)

### PR DESCRIPTION
**Pull request by ThomasDevoogdt**
#### ea2be408f27eb1009bbde7b0a2496018d4ca5e86
<pre>
REGRESSION: fix compilation if !ENABLE(VIDEO)

<a href="https://bugs.webkit.org/show_bug.cgi?id=271881">https://bugs.webkit.org/show_bug.cgi?id=271881</a>

Reviewed by Mike Wyrzykowski.

[WebCore]: FullscreenManager: add missing include

HTMLMediaElement normally includes HTMLMediaElementEnums,
but only if VIDEO is enabled.

[WebCore]: FullscreenManager: fix undefined reference to HTMLMediaElement

HTMLMediaElement is not compiled if VIDEO is not enabled.

[WebCore]: css: fix fix undefined reference to HTMLMediaElement

HTMLMediaElement is not compiled if VIDEO is not enabled.

[WebCore]: InspectorDOMAgent: fix fix undefined reference to HTMLMediaElement

HTMLMediaElement is not compiled if VIDEO is not enabled.

[WebKit]: WebGPU: drop duplicate directive in RemoteDevice

The #if PLATFORM(COCOA) &amp;&amp; ENABLE(VIDEO) directive is already defined
outside the function scope, so drop the inner ones.

[WebCore]: GPUDevice: fix undefined reference to HTMLMediaElement

HTMLMediaElement is not compiled if VIDEO is not enabled.

[WebKit]: WebFullScreenManagerProxy: add missing include

HTMLMediaElement normally includes HTMLMediaElementEnums,
but only if VIDEO is enabled.

[WebKit]: WebFullScreenManager: add missing include

HTMLMediaElement normally includes HTMLMediaElementEnums,
but only if VIDEO is enabled.

[WebKit]: InjectedBundlePageFullScreenClient: add missing include

HTMLMediaElement normally includes HTMLMediaElementEnums,
but only if VIDEO is enabled.

[WebKit]: WebCoreArgumentCoders.serialization.in: fix undefined reference

CaptionUserPreferencesDisplayMode and SerializedPlatformDataCueValue
are only compiled if VIDEO is enabled.

Signed-off-by: Thomas Devoogdt &lt;thomas@devoogdt.com&gt;
Canonical link: <a href="https://commits.webkit.org/276977@main">https://commits.webkit.org/276977@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/4a10cd053fe7d9ddd2fd549672a6a45a965319a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46301 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25437 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48892 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48974 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42342 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48608 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/29793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/22894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/37815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46879 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/29793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/39921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19037 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/29793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41019 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4345 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/29793 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41374 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50787 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/21305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/22894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/22597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/43929 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10248 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/22964 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22294 "Built successfully") | | |

### CODE REVIEW

**Review by mwyrzykowski — APPROVED** [review]

**Comment by ThomasDevoogdt** [comment]
@mwyrzykowski, @mcatanzaro is it possible to take this one further up? See the comment, a similar PR has been opened for 2.44, but perhaps it's better to first merge this one, and to do the cherry-picks manually? Let me know if any change is still required, or feel free to update the PR yourself.

FYI, this is part of the webkitgtk bump in buildroot that I prepared:
https://patchwork.ozlabs.org/project/buildroot/patch/20240401183930.2557159-2-thomas@devoogdt.com/

**Comment by mcatanzaro** [comment]
I see this has already received review approval, but it will fail to land because it contains multiple commits. So please squash all the commits into one. Then add the request-merge-queue label to request that it land. Leave a comment here when you do so, because I'm not sure that GitLab gives notifications for changes to labels.

Normally to request backport you can just add the GLib Suggested Backport label, but creating a merge request is fine too, especially if there are conflicts. The one commit rule still applies, though. Let me know in #26652 when it is ready.

Lastly, reminder that it's recommended to post a link to the pull request in the bug report (which would have been enough for us to notice this pull request).

**Comment by mcatanzaro** [comment]
> I'm not maintainer of the project, so I can't add any label. 

Huh, well that's quite a workflow problem, since the label is only useful if you're not a maintainer.

Well, OK then. Asking in a comment works just as well.

**Comment by mcatanzaro** [comment]
This doesn't quite follow the usual commit message format, but the first few lines do and I think those are all that merge-queue will care about. Let's find out....

**Comment by ThomasDevoogdt** [comment]
> The one commit rule still applies, though. Let me know in https://github.com/WebKit/WebKit/pull/26652 when it is ready.

I will update #26652 if this one got merged, and ping you. There were a few small merge conflicts. Will most probably be done tomorrow.

> This doesn't quite follow the usual commit message format, but the first few lines do and I think those are all that merge-queue will care about. Let's find out....

There are already a few other commits that got merged this way. It's not the first time I'm fixing stuff out here ;-)

**Pull request merged**
